js/core/utils/compare.js

```diff
@@ -0,0 +1,30 @@
+import domAdapter from "../dom_adapter";
+import coreDataUtils from "./data";
+import typeUtils from "./type";
+
+const valuesEqual = function(oldValue, newValue) {
```

Comment:
I suggest to move this method to `data` utility file. The `compare` filename looks too mysterious without a context.